sigar_unix.go

```diff
@@ -22,3 +27,43 @@ func (self *FileSystemUsage) Get(path string) error {
 
 	return nil
 }
+
+func (r *Rusage) Get(who int) error {
+	ru, err := getResourceUsage(who)
+	if err != nil {
+		return err
+	}
+
+	uTime := convertRtimeToDur(ru.Utime)
+	sTime := convertRtimeToDur(ru.Stime)
+
+	r.Utime = uTime
+	r.Stime = sTime
+	r.Maxrss = int64(ru.Maxrss)
+	r.Ixrss = int64(ru.Ixrss)
+	r.Idrss = int64(ru.Idrss)
+	r.Isrss = int64(ru.Isrss)
+	r.Minflt = int64(ru.Minflt)
+	r.Majflt = int64(ru.Majflt)
+	r.Nswap = int64(ru.Nswap)
+	r.Inblock = int64(ru.Inblock)
+	r.Oublock = int64(ru.Oublock)
+	r.Msgsnd = int64(ru.Msgsnd)
+	r.Msgrcv = int64(ru.Msgrcv)
+	r.Nsignals = int64(ru.Nsignals)
+	r.Nvcsw = int64(ru.Nvcsw)
+	r.Nivcsw = int64(ru.Nivcsw)
+
+	return nil
+}
+
+func getResourceUsage(who int) (unix.Rusage, error) {
+	r := unix.Rusage{}
+	err := unix.Getrusage(who, &r)
+
+	return r, err
+}
+
+func convertRtimeToDur(t unix.Timeval) time.Duration {
+	return time.Duration(t.Sec)*time.Second + time.Duration(t.Usec)*time.Microsecond
```

Comment:
`return time.Duration(t.Nano())`

Reply:
Unfortunately, `t` only has `Sec` and `Usec` attributes, because it is a `unix.Timeval`: https://godoc.org/golang.org/x/sys/unix#Timeval

Reply:
Are you sure? It's shown on the page you linked and here's the implementation.

https://github.com/golang/sys/blob/2c42eef0765b9837fbdab12011af7830f55f88f0/unix/timestruct.go#L79-L82